test_infra/stacks/glueray_stack.py

```diff
@@ -0,0 +1,116 @@
+from aws_cdk import aws_s3 as s3
+from aws_cdk import aws_s3_assets as s3_assets
+from constructs import Construct
+
+
+class GlueRayStack(Stack):  # type: ignore
+    def __init__(
+        self,
+        scope: Construct,
+        construct_id: str,
+        bucket: s3.Bucket,
+        **kwargs: str,
+    ) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+
+        self.script_bucket = s3.Bucket(
+            self,
+            "Script Bucket",
+            block_public_access=s3.BlockPublicAccess.BLOCK_ALL,
+            removal_policy=RemovalPolicy.DESTROY,
+        )
+
+        self.athena_workgroup = athena.CfnWorkGroup(
+            self,
+            "Workgroup",
+            state="ENABLED",
+            work_group_configuration=athena.CfnWorkGroup.WorkGroupConfigurationProperty(
+                result_configuration=athena.CfnWorkGroup.ResultConfigurationProperty(
+                    output_location=bucket.s3_url_for_object(""),
+                ),
+            ),
+            description="AWS SDK for pandas with Glue on Ray",
+            name="GlueRayWorkGroup",
+        )
+
+        self.glue_service_role = iam.Role(
+            self,
+            "Glue Job Role",
+            assumed_by=iam.ServicePrincipal("glue.amazonaws.com"),
+            managed_policies=[
+                iam.ManagedPolicy.from_aws_managed_policy_name("service-role/AWSGlueServiceRole"),
+                iam.ManagedPolicy.from_aws_managed_policy_name("CloudWatchFullAccess"),
+                iam.ManagedPolicy.from_aws_managed_policy_name("AmazonAthenaFullAccess"),
+            ],
+        )
+
+        bucket.grant_read_write(self.glue_service_role)
+        self.script_bucket.grant_read(self.glue_service_role)
+
+        # Grant data permissions to Glue job
+        amazon_reviews_pds_bucket = s3.Bucket.from_bucket_name(
+            self,
+            "amazon_reviews_pds",
+            bucket_name="amazon-reviews-pds",
+        )
+        amazon_reviews_pds_bucket.grant_read(self.glue_service_role)
+
+        # Where should the ZIP be uploaded
+        self.wrangler_asset_path = self.script_bucket.s3_url_for_object("awswrangler.zip")
+
+        # Create Glue Job
+        glue_job1 = self._create_glue_job(id="Glue Job 1", script_path="glue_scripts/glue_example_1.py")
+
+        # CFN outputs
+        CfnOutput(
+            self,
+            "AWS SDK for pandas ZIP Location",
+            value=self.wrangler_asset_path,
+            export_name="WranglerZipLocation",
+        )
+
+        CfnOutput(
+            self,
+            "Glue Job 1 Name",
+            value=glue_job1.ref,
+            export_name="GlueJob1Name",
+        )
+
+    def _deploy_glue_asset(self, id: str, path: str) -> s3_assets.Asset:
+        return s3_assets.Asset(
+            self,
+            id,
+            path=path,
+            readers=[self.glue_service_role],
+        )
+
+    def _create_glue_job(self, id: str, script_path: str) -> glue.CfnJob:
```

Comment:
Another approach to take would be to create the Glue job from a pytest fixture, and then delete it at the end of the session. An upside to this would be is that we could have multiple tests on the same Glue script at the same time without them clashing. However, a downside would be that it would be a lot harder to debug errors in the Glue job if it's deleted right after execution. The logs would still exist in CloudWatch, but finding them would not be intuitive.

Reply:
I think I have a slight preference for creating Glue jobs dynamically. It allows for more flexibility and less infra/stacks to manage + creating multiple Glue jobs with different scripts. It would also mean that we would not only test the script but also the ability to create the Glue job itself.

One way around the logs issue is to build a URL to the CW logs in the pytest fixture creating the job. This way all I would have to do when debugging is to follow that URL to find the logs.

That being said I am not sure how many Glue jobs we actually want to have within our testing. Perhaps the one is enough and the others would be within the Glue team infra

Reply:
+1 I agree with a slight preference to create jobs dynamically as we do for other transient resources such as glue databases.

Reply:
Would this mean we would include a zip and upload to s3 of `awswrangler.zip` as a fixture as well?

Reply:
I'm thinking about this but I would either need to:
1. Run the command line script which uploads `awswrangler.zip` to S3 from a Pytest fixture. This would be simple but it would essentially mean our tests only run on bash. For any contributor on Windows this would fail.
2. Re-create the whole functionality of building poetry, unzipping, zipping, and uploading to S3 with Python commands. This could get quite complex thought.

So I think for now I would want to keep this functionality in the CodeBuild.